### PR TITLE
Tweak tabtitle threshold w/ secondary icons on small tabs

### LINKED ISSUE #9466: Tab title isn't hidden for session tabs after the tab threshold is reached
## Test plan
See https://github.com/brave/browser-laptop/pull/9512

## Original issue description
- Did you search for similar issues before submitting this one?
Yes

- Describe the issue you encountered:
Tab title isn't hidden for session tabs after the tab threshold is reached

- Platform (Win7, 8, 10? macOS? Linux distro?):
All

- Brave Version (revision SHA):
Brave	0.17.1
rev	84dbc8e
Muon	4.0.3

- Steps to reproduce:
    1. Pin 8-10 tabs  on a clean install
    2. Open 20 tabs including a number of session tabs
    3. Tab title isn't hidden behind the session tab icon when the threshold size is reached 
    
- Actual result:
Tab title isn't hidden for session tabs after the tab threshold is reached

- Expected result:
Should hide text behind session tab icon once the threshold is reached

- Will the steps above reproduce in a fresh profile? If not what other info can be added?
Yes

- Is this an issue in the currently released version?
No

- Can this issue be consistently reproduced?
Yes

- Extra QA steps:
    1.
    2.
    3.

- Screenshot if needed:
image

- Any related issues:

### PR DESCRIPTION
**Pull request by cezaraugusto**
Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Auditors: @bsclifton
Close #9466

## Test Plan:
covered by automated tests
```
npm run test -- --grep="Tabs content - NewSessionIcon should not show icon if breakpoint is medium"
npm run test -- --grep="Tabs content - PrivateIcon should not show icon if breakpoint is medium"
npm run test -- --grep="Tabs content - Title should show text if breakpoint is mediumSmall and tab is not active"
npm run test -- --grep="Tabs content - Title should not show text if breakpoint is mediumSmall and tab is active"
```

## Manual Test:

1. Have enough private/new session/default tabs.
2. Resize window
3. Title should not truncate with secondary icons
4. Title should be aesthetically ok if there are no secondary icons

Reviewer Checklist:

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by bsclifton** [comment]
@cezaraugusto this is ready for review, right?

File: app/common/state/tabContentState.js [hunk 1]
```diff
@@ -152,7 +152,8 @@ const tabContentState = {
       // If closeIcon is fixed then there's no room for another icon
       !tabContentState.hasFixedCloseIcon(state, frameKey) &&
       // completely hide it for small sizes
-      !hasBreakpoint(frame.get('breakpoint'), ['mediumSmall', 'small', 'extraSmall', 'smallest'])
+      !hasBreakpoint(frame.get('breakpoint'),
+        ['medium', 'mediumSmall', 'small', 'extraSmall', 'smallest'])
```
Comment (NejcZdovc) [inline app/common/state/tabContentState.js]: I checkout my PR that refactored this function and I see that `medium` was not there before either. How come that we need to add it now? Was this working correctly before in previous versions or not?

Comment (cezaraugusto) [inline app/common/state/tabContentState.js]: the feature works as expected but turns out to have bad UI. It wasn't introduced in your PR.

**Comment by cezaraugusto** [comment]
@bsclifton yes ready. It wasn't a bug before and wasn't a regression as well but given the bad look and feel this tweak is needed. I'll label needs QA to double-check as well, thanks.

**Comment by NejcZdovc** [comment]
Not sure if this is part of this PR or not, but I think that it would be good to fix this one as well.

Open some empty new session tabs and try to resize. I think that we should display session icon with a number if there is no fav icon.

big:
image

medium:
image

small:
image

**Comment by cezaraugusto** [comment]
@NejcZdovc re https://github.com/brave/browser-laptop/pull/9512#issuecomment-309398392 I would go adding paper icon instead. It is the default for websites with broken favicons. Maybe just left the title, but I prefer the former. Anyway, your assumption is valid given empty tab is not pleasing.

/cc @bradleyrichter for the rescue

**Pull request merged**